src/cpio/client_providers/private_key_fetcher_provider/azure/azure_private_key_fetcher_provider_utils.h

```diff
@@ -31,7 +203,69 @@ class AzurePrivateKeyFetchingClientUtils {
   static void CreateHttpRequest(
       const PrivateKeyFetchingRequest& private_key_fetching_request,
       core::HttpRequest& http_request);
+
+  /**
+   * @brief Generate a new wrapping key
+   */
+  static std::pair<std::shared_ptr<EvpPkeyWrapper>,
+                   std::shared_ptr<EvpPkeyWrapper>>
+  GenerateWrappingKey();
+
+  /**
+   * @brief Convert a wrapping key in PEM
+   *
+   * @param wrappingKey RSA public key used to wrap a key.
+   */
+  static std::string EvpPkeyToPem(EVP_PKEY* wrappingKey);
+
+  /**
+   * @brief Convert a public PEM wrapping key to pkey
+   *
+   * @param wrappingPemKey RSA PEM key used to wrap a key.
+   */
+  static EVP_PKEY* GetPublicEvpPkey(std::string wrappingPemKey);
```

Comment:
Can you use the wrapper discussed [here](https://github.com/KenGordon/data-plane-shared-libraries/pull/13#discussion_r1541280678) (with smart pointer) in these function inputs and outputs please?

For this example:
```cpp
static std::shared_ptr<EvpPkeyWrapper> GetPublicEvpPkey(std::string wrappingPemKey);
```
Without the change, it's very unclear who is responsible to free `EVP_PKEY` in this example. It might be caller or it might be function itself because the function is static. You could document it, but smart pointer would be much nicer because you could easily know how the memory is managed.